### PR TITLE
Bug 2006067: Fix OLM error handling

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Errors were not being properly handled in a couple of OLM components. In one case, an error was not being passed to the setter function, and in another case, the right prop was not being passed to the error modal. The error modal and all the OLM models also looked like they weren't properly internationalized. I internationalized them and converted the erorr modal to TypeScript.

One error message I fixed looked pretty bad when I made it actually render the way the code was written to render it, so I spoke to Megan Hall about it:
Screen Shot 2021-11-02 at 5 10 37 PM

We added a check for whether users have the appropriate permissions (so we can avoid so-and-so-does-not-have-access issues here):
Screen Shot 2021-11-05 at 1 21 54 PM

We also moved any error messages into a modal to match the other one (text here is just an example; we won't hit this error anymore here with the access restriction):
Screen Shot 2021-11-05 at 12 49 26 PM

When you have insufficient permissions, you can't click approve or deny anymore here either:
Screen Shot 2021-11-05 at 1 21 24 PM

Hitting an error now gives you an error modal that has text in it (it was blank before). Text here is just an example; we won't hit this error anymore here with the access restriction: 
Screen Shot 2021-11-03 at 5 54 22 PM
Screen Shot 2021-11-05 at 12 58 26 PM

This bug appears to be present in 4.7 all the way through 4.10. We should figure out how far we're able to backport this. We should at least be able to backport the minimal change of making sure we're passing the error message down correctly to setters/modals.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2006067.

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:127]:
```suggestion
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:306]: We don't need both patch and update to modify an InstallPlan resource. Just one or the other. Discussed with @spadgett and he suggested we update to use `k8sPatch` everywhere.
```suggestion
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:316]:
```suggestion
      {needsApproval && canPatchInstallPlans && (
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:334]:
```suggestion
      {needsApproval && !canPatchInstallPlans && (
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:417]: Safer to use `Object.prototype.toString`:
```suggestion
      .catch((error) => errorModal({ error: error.toString() }));
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:433]:
```suggestion
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:118]:
```suggestion
```

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:415]:
```suggestion
    k8sPatch(InstallPlanModel, obj, [{ op: 'replace', path: '/spec/approved', value: true }])
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:136]:
```suggestion
        {(installObjIsInstallPlan && canPatchInstallPlans ||
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:155]:
```suggestion
        {!canPatchInstallPlans && installObjIsInstallPlan && (
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:346]:
```suggestion
      errorModal({ error: error.toString() });
```

**Review by TheRealJon — CHANGES_REQUESTED** [review]
Remove HOC pattern and update to use the same API verb everywhere when approving an install plan.

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:293]: Megan wants to preserve the user name aspect, but I will use the hooks instead! Seems like it's much more readable.

**Comment by rebeccaalpert** [comment]
Addressed PR feedback. e2e test failure appears to be a flake.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:117]: This is also declared in `frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx`

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:278]: We should probably align to the error message formulation from the api-server
```suggestion
        'olm~User "{{user}}" cannot patch resource InstallPlans in API group "{{apiGroup}}" in the namespace "{{namespace}}."',
```
Screenshot 2021-11-18 at 23 21 03

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:9]: 👍

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:351]: Where are we using these?

Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/locales/en/olm.json:351]: Kebab menus and breadcrumbs, mostly. I noticed one of the OLM terms wasn't internationalized on one of the OLM pages and the cause was that we didn't have i18n labelKeys on the model. I went ahead and did them all just to be safe.

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:278]: Just checking with Megan to make sure she's cool with it!

Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:278]: Hey @jhadvig - Megan was hoping to keep this message since it provides the "why" to the user. Making messaging clear to the users is a priority for the UXD team. If it's a goal to keep the UI and API messaging similar, would we be able to open a bug to add more details to the API message instead?

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:117]: Is it better practice to declare it once and export it? I haven't worked with these very often.

File: frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx
Comment (jhadvig) [reply frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:278]: @rebeccaalpert if this message if OK by the UXD team them Im fine ! 👍

Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx:278]: Okay cool; thank you! It means a lot to Megan.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (jhadvig) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:117]: yes, we should make our code DRY (Dont Repeat Yourself)

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (jhadvig) [reply frontend/packages/operator-lifecycle-manager/locales/en/olm.json:351]: got it !

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx
Comment (rebeccaalpert) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:117]: I gave it a go, but it looks like since it's a hook, it will violate the rules of hooks if I turn it into a shared function.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Review by jhadvig — APPROVED** [review]

**Comment by rebeccaalpert** [comment]
Putting on hold until merge queue is fixed.

**Comment by rebeccaalpert** [comment]
It seems we now have the "Unauthorized" flake for the pseudolocalization tests.

**Comment by rebeccaalpert** [comment]
Manual cherry-pick it is!

**Pull request merged**
